piz/src/spec.rs

```diff
@@ -547,6 +550,19 @@ impl<'a> FileMetadata<'a> {
     }
 }
 
+fn parse_msdos(time: u16, date: u16) -> NaiveDateTime {
+    let seconds = (0b0000_0000_0001_1111 & time) as u32 * 2; // MSDOS uses 2-second precision
+    let minutes = (0b0000_0111_1110_0000 & time) as u32 >> 5;
+    let hours = (0b1111_1000_0000_0000 & time) as u32 >> 11;
+
+    let days = 0b0000_0000_0001_1111 & date as u32;
```

Comment:
Nit: parens to match the other lines? (Though it looks like it compiles to equivalent code regardless: https://rust.godbolt.org/z/GnnYnT)